src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.inline.hpp

```diff
@@ -44,43 +44,30 @@ HeapWord* ShenandoahHeapRegion::allocate_aligned(size_t size, ShenandoahAllocReq
 
   HeapWord* orig_top = top();
   size_t addr_as_int = (uintptr_t) orig_top;
+  size_t alignment_in_words = alignment_in_bytes / HeapWordSize;
 
-  // unalignment_bytes is the amount by which current top() exceeds the desired alignment point.  We subtract this amount
-  // from alignment_in_bytes to determine padding required to next alignment point.
+  // unalignment_words is the amount by which current top() exceeds the desired alignment point.  We subtract this amount
+  // from alignment_in_words to determine padding required to next alignment point.
 
-  // top is HeapWord-aligned so unalignment_bytes is a multiple of HeapWordSize
-  size_t unalignment_bytes = addr_as_int % alignment_in_bytes;
-  size_t unalignment_words = unalignment_bytes / HeapWordSize;
-
-  size_t pad_words;
-  HeapWord* aligned_obj;
-  if (unalignment_words > 0) {
-    pad_words = (alignment_in_bytes / HeapWordSize) - unalignment_words;
+  HeapWord* aligned_obj = (HeapWord*) align_up(orig_top, alignment_in_bytes);
```

Comment:
I think this method's declaration `allocate_aligned(...)` in the `.hpp` uses `alignment_in_words` as the name of the second parm. Similarly the comment in the header. Those should be corrected too to say `alignment_in_bytes`.

Reply:
We do know that this is alignment must be a mutiple of HeapWordSize (as demonstrated by the integer division you do further above). I'd just assert for the sake of peace of mind from an API perspective that this is always so by:
```
assert(is_aligned(alignment_in_bytes, HeapWordSize), "Expected heap word alignment");
```

Reply:
Good catch.  Correcting the name in the header and in comment.  Adding the suggested assert.